core/embed/lib/image.c

```diff
@@ -224,12 +224,13 @@ secbool check_image_contents(const image_header *const hdr, uint32_t firstskip,
     return secfalse;
   }
 
-  const void *data =
-      flash_area_get_address(area, firstskip, IMAGE_CHUNK_SIZE - firstskip);
+  int remaining = hdr->codelen;
+
+  const void *data = flash_area_get_address(
```

Comment:
why is `flash_area_get_address` affected by `remaining`? shouldn't `IMAGE_CHUNK_SIZE` always match?

Reply:
the area can be smaller than the chunk:
- on u5, we will (maybe) have 128kB bootloader and 256kb chunks
- also the fw area is not divisble by 256kb, so the last part will always by smaller